File: src/levels/L10-observer/ts/strategy.ts
class Order {
  private status = 'pendiente'

  constructor(private notifier: Subscriber) {}

  // Strategy elige UNA forma de avisar: pantalla, o push, o lealtad.
  // region: Order#complete!
  complete() {
    this.status = 'listo'
    this.notifier.update(this)
  }
  // endregion
}

File: src/levels/L10-observer/strategy.rb
class Order
  def initialize(notifier) = @notifier = notifier

  # Strategy elige UNA forma de avisar: pantalla, o push, o lealtad.
  # region: Order#complete!
  def complete!
    @status = :listo
    @notifier.update(self)
  end
  # endregion
end
